### PR TITLE
Feat/#125 edit proxy failover

### PR DESCRIPTION
**Pull request by agobeljic1**
## Issue

https://graviteecommunity.atlassian.net/browse/APIM-126

## Description

Edit API proxy failover

## Additional context

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/feat-125-edit-proxy-failover/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-izsshadftv.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/proxy/apis.proxy.route.html [hunk 1]
```diff
@@ -77,7 +77,7 @@
           permission
           permission-only="['api-definition-r']"
           ui-sref-active="aui-nav-selected"
-          ui-sref="management.apis.detail.proxy.failover"
+          ui-sref="management.apis.detail.proxy.ng-failover"
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/proxy/apis.proxy.route.html]: If you chnage the ui-sref I would remove the old controller and template.

If you keep the old controller and template I would let the old ui-sref. Then after validation from Antoine I would remove everything and do this change. 

You can do whatever you want both solutions are ok for me even if I prefer the second one personally. Wdyt ?

**Review by Okhelifi — COMMENTED** [review]
Well done LGTM for me. 
I just have one comment and I would rephrase the commit message for something like 
feat(console): migrate api proxy failover page
feat(console): ....

File: gravitee-apim-console-webui/src/management/api/proxy/apis.proxy.route.html (hunk 1 above)
Comment (agobeljic1) [reply gravitee-apim-console-webui/src/management/api/proxy/apis.proxy.route.html]: I will then keep the old sref, until validation passes

**Review by Okhelifi — COMMENTED** [review]
Good for me

**Review by Okhelifi — APPROVED** [review]

File: gravitee-apim-console-webui/src/management/api/proxy/failover/api-proxy-failover.component.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+      <!-- Max Attempts -->
+      <div class="failover-card__max-attempts" [class.disabled]="!enabled.value">
+        <div class="failover-card__max-attempts__label">Max attempts</div>
+        <p class="failover-card__max-attempts__hint">
+          Limits the number of possible tries before returning an error. Each try get an endpoint according to the load balancing algorithm.
+        </p>
+        <mat-form-field class="failover-card__max-attempts__form-field" appearance="fill">
+          <mat-label>Max Attempts</mat-label>
+          <input matInput formControlName="maxAttempts" type="number" min="1" />
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/proxy/failover/api-proxy-failover.component.html]: What do you think to add required to have the `*` on the label ? 
And add `mat-error` if the required validator throw 

same for all required field.

File: gravitee-apim-console-webui/src/management/api/proxy/failover/api-proxy-failover.component.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        catchError(({ error }) => {
+          this.snackBarService.error(error.message);
+          return EMPTY;
+        }),
+        tap(() => this.ngOnInit()),
+      )
+      .subscribe();
+  }
+
+  public get enabled() {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/proxy/failover/api-proxy-failover.component.ts]: I often place the getter on top of the controller. How about adding the `.value` here to avoid having it in the template. WDYT ?

**Review by ThibaudAV — APPROVED** [review]
In my opinion. You can group into a single commit.
I think that in a single PR you should keep the commits as readable as possible. Each one should be the next build step. but not a correction of the previous commit. in this case we can just modify the previous commit (with `!fixup` for ex)

And FIY we try to respect https://www.conventionalcommits.org . (on more recent projects a linter will validate it)
So commit never start with uppercase 😁 

Otherwise. Sounds great 🚀 🚀 🚀

File: gravitee-apim-console-webui/src/management/api/proxy/failover/api-proxy-failover.component.ts (hunk 3 above)
Comment (agobeljic1) [reply gravitee-apim-console-webui/src/management/api/proxy/failover/api-proxy-failover.component.ts]: We can put it on top.

I thought to put it here without the .value so I can handle this.enabled.valueChanges.... with it also

File: gravitee-apim-console-webui/src/management/api/proxy/failover/api-proxy-failover.component.html (hunk 2 above)
Comment (agobeljic1) [reply gravitee-apim-console-webui/src/management/api/proxy/failover/api-proxy-failover.component.html]: I added mat-error for fields.

I disagree with the * since in angular 13 they added stars to all required fields. If we will migrate in the near future, maybe it doesn't make much sense
